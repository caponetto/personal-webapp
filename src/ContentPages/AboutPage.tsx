import Paper from "@mui/material/Paper";
import Box from "@mui/material/Box";
import React from "react";
import Grid from "@mui/material/Grid";
import Fade from "@mui/material/Fade";

export function AboutPage() {
  return (
    <Box>
      <Grid container spacing={2}>
        <Fade in={true} timeout={100}>
          <Grid item xs={12}>
            <Paper sx={{ height: "200px" }} elevation={2}>
              TODO
            </Paper>
          </Grid>
        </Fade>
        <Fade in={true} timeout={500}>
          <Grid item xs={12} lg={6}>
            <Paper sx={{ height: "200px" }} elevation={2}>
              TODO
            </Paper>
          </Grid>
        </Fade>
        <Fade in={true} timeout={1000}>
          <Grid item xs={12} lg={6}>
            <Paper sx={{ height: "200px" }} elevation={2}>
              TODO
            </Paper>
          </Grid>
        </Fade>
        <Fade in={true} timeout={1500}>
          <Grid item xs={12}>
            <Paper sx={{ height: "200px" }} elevation={2}>
              TODO: cloud of words
            </Paper>
          </Grid>
        </Fade>
      </Grid>
    </Box>
  );
}
