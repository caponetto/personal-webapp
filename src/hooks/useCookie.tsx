import Cookies, { CookieAttributes } from "js-cookie";
import { useCallback, useMemo, useState } from "react";

export type CookieNames = "color_mode" | "show_snackbar";

export function useCookie<T>(name: CookieNames, defaultValue: T, options?: CookieAttributes): [T, (v: T) => void] {
  const defaultOptions = useMemo<CookieAttributes>(
    () => ({
      expires: 100, // days
      path: "/",
      sameSite: "strict",
    }),
    [],
  );

  const [value, setValue] = useState(() => {
    try {
      const cookie = Cookies.get(name);
      if (cookie) {
        return JSON.parse(cookie) as T;
      }
    } catch (e) {
      console.error(`Cannot parse cookie "${name}"; using default value "${defaultValue}" instead`);
    }
    Cookies.set(name, JSON.stringify(defaultValue), options || defaultOptions);
    return defaultValue;
  });

  const updateCookie = useCallback(
    (newValue: T) => {
      Cookies.set(name, JSON.stringify(newValue), options ?? defaultOptions);
      setValue(newValue);
    },
    [defaultOptions, name, options],
  );

  return [value, updateCookie];
}
