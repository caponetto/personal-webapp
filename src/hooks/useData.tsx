import { useMemo } from "react";
import { useTranslation } from "react-i18next";
import { AppData } from "../data";
import { routes } from "../routes";

export function useData() {
  const { t } = useTranslation();

  const locations = useMemo(
    () => ({
      unicamp: {
        name: t("literal:unicamp"),
        url: routes.urls.unicamp,
      },
      unifei: {
        name: t("literal:unifei"),
        url: routes.urls.unifei,
      },
      udacity: {
        name: t("literal:udacity"),
        url: routes.urls.udacity,
      },
      oracle: {
        name: t("literal:oracle"),
        url: routes.urls.oracle,
      },
      redHat: {
        name: t("literal:redHat"),
        url: routes.urls.redHat,
      },
      samsung: {
        name: t("literal:samsung"),
        url: routes.urls.samsung,
      },
      iFood: {
        name: t("literal:iFood"),
        url: routes.urls.iFood,
      },
      motorola: {
        name: t("literal:motorola"),
        url: routes.urls.motorola,
      },
      b2ml: {
        name: t("literal:b2mlSystems"),
        url: routes.urls.b2ml,
      },
    }),
    [t]
  );

  const personal = useMemo(
    () => ({
      fullName: t("personal:fullName"),
      location: {
        country: t("literal:brazil"),
        flag: "ðŸ‡§ðŸ‡·",
        url: routes.urls.mapsCountry,
      },
    }),
    [t]
  );

  const about = useMemo(
    () => ({
      welcome: t("about:welcome"),
      paragraphs: Array.from(t("about:paragraphs", { returnObjects: true })),
    }),
    [t]
  );

  const journey = useMemo(
    () => ({
      education: [
        {
          title: t("journey:education.master.title"),
          period: {
            start: 2017,
          },
          location: locations.unicamp,
        },
        {
          title: t("journey:education.bachelor.title"),
          period: {
            start: 2011,
          },
          location: locations.unifei,
        },
      ],
      certification: [
        {
          title: t("literal:computerVisionNanoDegree"),
          period: {
            start: 2019,
          },
          location: locations.udacity,
        },
        {
          title: t("literal:oracleCertifiedProfessionalJavaSix"),
          period: {
            start: 2012,
          },
          location: locations.oracle,
        },
      ],
      experience: [
        {
          title: `${t("literal:softwareEngineer")} â†’ ${t("literal:seniorSoftwareEngineer")}`,
          period: {
            start: 2019,
            end: t("literal:present"),
          },
          location: locations.redHat,
        },
        {
          title: `${t("literal:softwareEngineer")} â†’ ${t("literal:seniorSoftwareEngineer")}`,
          period: {
            start: 2017,
            end: 2019,
          },
          location: locations.samsung,
        },
        {
          title: t("literal:softwareEngineer"),
          period: {
            start: 2017,
            end: 2017,
          },
          location: locations.iFood,
        },
        {
          title: t("literal:graduateResearcher"),
          period: {
            start: 2015,
            end: 2017,
          },
          location: locations.unicamp,
        },
        {
          title: t("literal:juniorSoftwareEngineer"),
          period: {
            start: 2012,
            end: 2015,
          },
          location: locations.samsung,
        },
        {
          title: t("literal:softwareDevelopmentIntern"),
          period: {
            start: 2011,
            end: 2011,
          },
          location: locations.motorola,
        },
        {
          title: t("literal:undergraduateResearcher"),
          period: {
            start: 2007,
            end: 2010,
          },
          location: locations.unifei,
        },
        {
          title: t("literal:softwareDevelopmentIntern"),
          period: {
            start: 2007,
            end: 2007,
          },
          location: locations.b2ml,
        },
      ],
      toolbox: [
        t("literal:amazonWebServices"),
        t("literal:android"),
        t("literal:apacheKafka"),
        t("literal:architecture"),
        t("literal:backend"),
        t("literal:csharp"),
        t("literal:cpp"),
        t("literal:css"),
        t("literal:classification"),
        t("literal:clustering"),
        t("literal:computerVision"),
        t("literal:cicd"),
        t("literal:deepLearning"),
        t("literal:designPatterns"),
        t("literal:docker"),
        t("literal:frontend"),
        t("literal:gwt"),
        t("literal:git"),
        t("literal:html"),
        t("literal:jboss"),
        t("literal:jni"),
        t("literal:jpa"),
        t("literal:java"),
        t("literal:javaScript"),
        t("literal:kanban"),
        t("literal:matlab"),
        t("literal:machineLearning"),
        t("literal:multiTaskLearning"),
        t("literal:neuralNetworks"),
        t("literal:noSql"),
        t("literal:nodeJs"),
        t("literal:oop"),
        t("literal:openShift"),
        t("literal:optimization"),
        t("literal:python"),
        t("literal:quarkus"),
        t("literal:restfulWs"),
        t("literal:reactJs"),
        t("literal:recommenderSystems"),
        t("literal:regression"),
        t("literal:scrum"),
        t("literal:sql"),
        t("literal:speechEnhancement"),
        t("literal:springFramework"),
        t("literal:systemDesign"),
        t("literal:tdd"),
        t("literal:tensorFlow"),
        t("literal:tomcat"),
        t("literal:typeScript"),
        t("literal:universalWindowsPlatform"),
        t("literal:xmpp"),
      ],
    }),
    [t, locations]
  );

  const text = useMemo(
    () => ({
      mastersThesis: [
        {
          type: "thesis",
          title: t("text:mastersThesis"),
          releaseDate: new Date("Oct 27, 2017"),
          publishedAt: "UNICAMP",
          keywordKeys: ["machineLearning", "multiTaskLearning", "clustering"],
          url: routes.urls.mastersThesis,
        },
      ],
      blogPosts: [
        {
          type: "post",
          title: t("text:blogPosts.workspaceCollaboration"),
          releaseDate: new Date("Sep 25, 2019"),
          publishedAt: "KIE Community",
          keywordKeys: ["businessCentral", "git", "changeRequests"],
          url: routes.urls.blogPosts.workspaceCollaboration,
        },
        {
          type: "post",
          title: t("text:blogPosts.playingWithObjectDetection"),
          releaseDate: new Date("Oct 11, 2019"),
          publishedAt: "Towards Data Science",
          keywordKeys: ["machineLearning", "computerVision", "objectDetection"],
          url: routes.urls.blogPosts.playingWithObjectDetection,
        },
        {
          type: "post",
          title: t("text:blogPosts.randomVsGridSearch"),
          releaseDate: new Date("Nov 14, 2019"),
          publishedAt: "Towards Data Science",
          keywordKeys: ["machineLearning", "optimization", "randomSearch"],
          url: routes.urls.blogPosts.randomVsGridSearch,
        },
        {
          type: "post",
          title: t("text:blogPosts.mavenArchetypesInBC"),
          releaseDate: new Date("Jan 21, 2020"),
          publishedAt: "KIE Community",
          keywordKeys: ["businessCentral", "maven", "archetypes"],
          url: routes.urls.blogPosts.mavenArchetypesInBC,
        },
        {
          type: "post",
          title: t("text:blogPosts.emptyReposInBC"),
          releaseDate: new Date("Feb 28, 2020"),
          publishedAt: "KIE Community",
          keywordKeys: ["businessCentral", "git", "maven"],
          url: routes.urls.blogPosts.emptyReposInBC,
        },
        {
          type: "post",
          title: t("text:blogPosts.squashCommitsInBC"),
          releaseDate: new Date("Feb 28, 2020"),
          publishedAt: "KIE Community",
          keywordKeys: ["businessCentral", "git", "changeRequests"],
          url: routes.urls.blogPosts.squashCommitsInBC,
        },
        {
          type: "post",
          title: t("text:blogPosts.improvedWorkflowInBC"),
          releaseDate: new Date("Mar 31, 2020"),
          publishedAt: "KIE Community",
          keywordKeys: ["businessCentral", "git", "syncRepositories"],
          url: routes.urls.blogPosts.improvedWorkflowInBC,
        },
        {
          type: "post",
          title: t("text:blogPosts.exportAsGist"),
          releaseDate: new Date("Apr 9, 2020"),
          publishedAt: "KIE Community",
          keywordKeys: ["kieTools", "github", "dmn", "bpmn"],
          url: routes.urls.blogPosts.exportAsGist,
        },
        {
          type: "post",
          title: t("text:blogPosts.backendKogitoTooling"),
          releaseDate: new Date("Sept 22, 2020"),
          publishedAt: "KIE Community",
          keywordKeys: ["kieTools", "backend", "vsCode"],
          url: routes.urls.blogPosts.backendKogitoTooling,
        },
        {
          type: "post",
          title: t("text:blogPosts.authoringOnGitpod"),
          releaseDate: new Date("Aug 9, 2021"),
          publishedAt: "KIE Community",
          keywordKeys: ["kieTools", "gitpod", "bpmn", "dmn"],
          url: routes.urls.blogPosts.authoringOnGitpod,
        },
        {
          type: "post",
          title: t("text:blogPosts.dmnDevSandbox"),
          releaseDate: new Date("Aug 23, 2021"),
          publishedAt: "KIE Community",
          keywordKeys: ["kieTools", "dmn", "openShift"],
          url: routes.urls.blogPosts.dmnDevSandbox,
        },
        {
          type: "post",
          title: t("text:blogPosts.teachableMachineBitingNails"),
          releaseDate: new Date("Jan 10, 2022"),
          publishedAt: "Towards Data Science",
          keywordKeys: ["teachableMachine", "reactJs", "openShift"],
          url: routes.urls.blogPosts.teachableMachineBitingNails,
        },
        {
          type: "post",
          title: t("text:blogPosts.deployKieSandboxToOpenshift"),
          releaseDate: new Date("Jan 27, 2022"),
          publishedAt: "KIE Community",
          keywordKeys: ["kieTools", "openShift"],
          url: routes.urls.blogPosts.deployKieSandboxToOpenshift,
        },
      ],
    }),
    [t]
  );

  const talk = useMemo(
    () => ({
      lives: [
        {
          type: "live",
          title: t("talk:lives.workspaceCollaborationInBC"),
          releaseDate: new Date("Sept 8, 2020"),
          publishedAt: "KIE Community",
          keywordKeys: ["businessCentral", "git", "changeRequests"],
          url: routes.urls.lives.workspaceCollaborationInBC,
        },
        {
          type: "live",
          title: t("talk:lives.mavenArchetypesInBC"),
          releaseDate: new Date("Sept 15, 2020"),
          publishedAt: "KIE Community",
          keywordKeys: ["businessCentral", "maven", "archetypes"],
          url: routes.urls.lives.mavenArchetypesInBC,
        },
        {
          type: "live",
          title: t("talk:lives.streamDecisionsWithDmnAndKafka"),
          releaseDate: new Date("Jun 8, 2021"),
          publishedAt: "KIE Community",
          keywordKeys: ["dmn", "openShift", "apacheKafka"],
          url: routes.urls.lives.streamDecisionsWithDmnAndKafka,
        },
      ],
      conferences: [
        {
          type: "conference",
          title: t("talk:conferences.streamDecisionsWithDmnAndKafka"),
          releaseDate: new Date("Jun 8, 2021"),
          publishedAt: "The Developer's Conference",
          keywordKeys: ["dmn", "openShift", "apacheKafka"],
          url: routes.slides.streamingDmnKafka,
        },
      ],
    }),
    [t]
  );

  const code = useMemo(
    () => ({
      repositories: [
        {
          type: "code",
          title: t("code:repositories.bhc"),
          releaseDate: new Date("Feb 16, 2021"),
          publishedAt: "GitHub",
          keywordKeys: ["python", "clustering", "bayesianStatistics"],
          url: routes.urls.repositories.bhc,
        },
        {
          type: "code",
          title: t("code:repositories.vsCodeQuarkusDJL"),
          releaseDate: new Date("Oct 16, 2020"),
          publishedAt: "GitHub",
          keywordKeys: ["computerVision", "quarkus", "vsCodeExtension", "typeScript"],
          url: routes.urls.repositories.vsCodeQuarkusDJL,
        },
        {
          type: "code",
          title: t("code:repositories.lessIsMore"),
          releaseDate: new Date("Apr 10, 2021"),
          publishedAt: "GitHub",
          keywordKeys: ["artifact", "gitHubActions", "typeScript"],
          url: routes.urls.repositories.lessIsMore,
        },
        {
          type: "code",
          title: t("code:repositories.jupyterNotebooks"),
          releaseDate: new Date("Oct 8, 2019"),
          publishedAt: "GitHub",
          keywordKeys: ["machineLearning", "jupyterNotebooks"],
          url: routes.urls.repositories.jupyterNotebooks,
        },
        {
          type: "code",
          title: t("code:repositories.vsCodeDiffViewer"),
          releaseDate: new Date("Feb 23, 2021"),
          publishedAt: "GitHub",
          keywordKeys: ["vsCodeExtension", "gitDiff", "typeScript"],
          url: routes.urls.repositories.vsCodeDiffViewer,
        },
        {
          type: "code",
          title: t("code:repositories.vsCodeToxicityClassifier"),
          releaseDate: new Date("Sept 19, 2020"),
          publishedAt: "GitHub",
          keywordKeys: ["vsCodeExtension", "tensorFlowJs", "typeScript"],
          url: routes.urls.repositories.vsCodeToxicityClassifier,
        },
        {
          type: "code",
          title: t("code:repositories.streamDecisionsWithDmnAndKafka"),
          releaseDate: new Date("Jun 2, 2021"),
          publishedAt: "GitHub",
          keywordKeys: ["apacheKafka", "dmn", "docker", "quarkus"],
          url: routes.urls.repositories.streamDecisionsWithDmnAndKafka,
        },
        {
          type: "code",
          title: t("code:repositories.vsCodeObjectDetection"),
          releaseDate: new Date("Jul 4, 2020"),
          publishedAt: "GitHub",
          keywordKeys: ["vsCodeExtension", "typeScript", "tensorFlowJs"],
          url: routes.urls.repositories.vsCodeObjectDetection,
        },
        {
          type: "code",
          title: t("code:repositories.personalWebApp"),
          releaseDate: new Date("Oct 24, 2021"),
          publishedAt: "GitHub",
          keywordKeys: ["reactJs", "typeScript", "webApp"],
          url: routes.urls.repositories.personalWebApp,
        },
        {
          type: "code",
          title: t("code:repositories.teachableMachinePlayground"),
          releaseDate: new Date("Jan 9, 2022"),
          publishedAt: "GitHub",
          keywordKeys: ["teachableMachine", "reactJs", "openShift"],
          url: routes.urls.repositories.teachableMachinePlayground,
        },
      ],
    }),
    [t]
  );

  return useMemo(
    () =>
      ({
        personal,
        about,
        journey,
        text,
        talk,
        code,
      } as AppData),
    [about, code, journey, personal, talk, text]
  );
}
