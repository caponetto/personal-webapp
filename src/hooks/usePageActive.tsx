import { useState } from "react";
import { useTimeout } from "./useTimeout";

export function usePageActive() {
  const [active, setActive] = useState(false);
  useTimeout(() => setActive(true), 0);
  return active;
}
