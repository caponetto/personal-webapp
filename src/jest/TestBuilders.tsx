import * as router from "react-router";
import { MediaItem } from "../schema";

export const DEFAULT_KEYWORD_SELECTION = {
  selectionMap: new Map(),
  onToggleSelection: jest.fn(),
  onClearSelection: jest.fn(),
  isAnySelected: false,
};

export const createMediaItem = (args: Partial<MediaItem>): MediaItem => ({
  title: args.title ?? "Title",
  keywordKeys: args.keywordKeys ?? [],
  kind: args.kind ?? "post",
  releaseDate: args.releaseDate ?? new Date(),
  publication: args.publication ?? "gitHub",
});

export const mockLocationTo = (args: Partial<router.Location>) => {
  jest.spyOn(router, "useLocation").mockImplementation(() => ({
    pathname: args.pathname ?? "/",
    search: args.search ?? "",
    hash: args.hash ?? "#",
    key: args.key ?? "default",
    state: args.state ?? undefined,
  }));
};
