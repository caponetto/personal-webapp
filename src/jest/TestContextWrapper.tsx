import { ReactElement } from "react";
import { I18nextProvider } from "react-i18next";
import { AppContextProvider } from "../context/AppContextProvider";
import i18n from "./I18nForTests";

export function usingTestingI18nContext(children: ReactElement) {
  return {
    wrapper: <I18nextProvider i18n={i18n}>{children}</I18nextProvider>,
  };
}

export function usingTestingAppContext(children: ReactElement) {
  return {
    wrapper: <AppContextProvider>{children}</AppContextProvider>,
  };
}

export function usingTestingContext(children: ReactElement) {
  return usingTestingI18nContext(usingTestingAppContext(children).wrapper).wrapper;
}
