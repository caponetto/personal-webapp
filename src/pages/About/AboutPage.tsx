import Box from "@mui/material/Box";
import CardActions from "@mui/material/CardActions";
import Fade from "@mui/material/Fade";
import Stack from "@mui/material/Stack";
import Typography from "@mui/material/Typography";
import { HoverableCard } from "../../components/card";
import { Page } from "../../components/page";
import { SocialBar } from "../../components/social";
import { useApp } from "../../context/AppContext";
import { Fonts } from "../../fonts";
import { Paragraph } from "./Paragraph";

const FADE_TIME = {
  card: 500,
};

export default function AboutPage() {
  const {
    schema: { personal, about },
  } = useApp();

  return (
    <Page name={"about"}>
      <Fade in={true} timeout={FADE_TIME.card}>
        <div>
          <HoverableCard sx={{ p: "28px", width: { xs: "100%", xl: "1170px" }, margin: "0 auto" }}>
            <Stack spacing={4}>
              <Typography
                component="div"
                sx={{
                  textAlign: "center",
                  fontWeight: "bold",
                  fontFamily: Fonts.PHILOSOPHER,
                  fontSize: { xs: "16px", sm: "20px" },
                }}
              >
                {about.welcome}
              </Typography>
              {about.paragraphs.map((paragraph: string, idx: number) => (
                <Paragraph key={`about-paragraph-${idx}`} content={paragraph} />
              ))}
            </Stack>
            <CardActions disableSpacing sx={{ display: { xs: "auto", md: "none" } }}>
              <Box sx={{ width: 1 }}>
                <SocialBar sx={{ width: "150px", mx: "auto", float: "center" }} urls={personal.urls} />
              </Box>
            </CardActions>
          </HoverableCard>
        </div>
      </Fade>
    </Page>
  );
}
