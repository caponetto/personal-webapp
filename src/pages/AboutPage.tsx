import Box from "@mui/material/Box";
import CardActions from "@mui/material/CardActions";
import Fade from "@mui/material/Fade";
import Stack from "@mui/material/Stack";
import Typography from "@mui/material/Typography";
import { ReactNode } from "react";
import { HoverableCard } from "../components/card";
import { Page } from "../components/page";
import { SocialBar } from "../components/social";
import { useApp } from "../context/AppContext";
import { Fonts } from "../fonts";
import { usePageActive } from "../hooks/usePageActive";
import { routes } from "../routes";

const FADE_TIME = {
  card: 500,
};

export default function AboutPage() {
  const app = useApp();
  const active = usePageActive(0);

  return (
    <Page name={"about"}>
      {active && (
        <Fade in={true} timeout={FADE_TIME.card}>
          <div>
            <HoverableCard sx={{ p: "28px", width: { xs: "100%", xl: "1170px" }, margin: "0 auto" }}>
              <Stack spacing={4}>
                <Typography
                  component="div"
                  sx={{
                    textAlign: "center",
                    fontWeight: "bold",
                    fontFamily: Fonts.PHILOSOPHER,
                    fontSize: { xs: "16px", sm: "20px" },
                  }}
                >
                  {app.schema.about.welcome}
                </Typography>
                {app.schema.about.paragraphs.map((paragraph: string, idx: number) => (
                  <Paragraph key={`about-paragraph-${idx}`}>{paragraph}</Paragraph>
                ))}
              </Stack>
              <CardActions disableSpacing sx={{ display: { xs: "auto", md: "none" } }}>
                <Box sx={{ width: 1 }}>
                  <SocialBar sx={{ width: "150px", mx: "auto", float: "center" }} urls={routes.urls.social} />
                </Box>
              </CardActions>
            </HoverableCard>
          </div>
        </Fade>
      )}
    </Page>
  );
}

function Paragraph(props: { children: ReactNode }) {
  return (
    <Typography component="div">
      <Box
        sx={{
          textAlign: "justify",
          lineHeight: 1.7,
          fontFamily: Fonts.NUNITO,
          fontSize: { xs: "13px", sm: "16px" },
        }}
      >
        {props.children}
      </Box>
    </Typography>
  );
}
