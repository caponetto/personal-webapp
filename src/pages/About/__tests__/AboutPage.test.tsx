import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import * as router from "react-router";
import { Router } from "react-router";
import { usePageActive } from "../../../hooks/usePageActive";
import { mockLocationTo } from "../../../jest/TestBuilders";
import { usingTestingContext } from "../../../jest/TestContextWrapper";
import AboutPage from "../AboutPage";

jest.spyOn(router, "useNavigate").mockReturnValue(jest.fn());
jest.mock("../../../hooks/usePageActive");
const mockUsePageActive = usePageActive as jest.MockedFunction<typeof usePageActive>;

describe("AboutPage", () => {
  it("should match the snapshot", () => {
    const pageName = "about";
    mockLocationTo({ pathname: `/${pageName}` });
    mockUsePageActive.mockReturnValue(true);
    const history = createMemoryHistory({ initialEntries: [`/${pageName}`] });
    const { getByTestId } = render(
      usingTestingContext(
        <Router location={history.location} navigator={history}>
          <AboutPage />
        </Router>,
      ),
    );
    expect(getByTestId(`${pageName}-page`)).toMatchSnapshot();
  });
});
