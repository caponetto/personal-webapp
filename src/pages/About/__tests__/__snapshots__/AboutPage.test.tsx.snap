// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage should match the snapshot 1`] = `
<div
  class="MuiBox-root css-zcrwk5"
  data-testid="about-page"
>
  <div
    style="opacity: 1; webkit-transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation2 MuiCard-root css-eegc1c-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiStack-root css-8mcjw0-MuiStack-root"
      >
        <div
          class="MuiTypography-root MuiTypography-body1 css-efiysx-MuiTypography-root"
        >
          welcome
        </div>
        <div
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-xgar5s"
          >
            p
          </div>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-xgar5s"
          >
            a
          </div>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-xgar5s"
          >
            r
          </div>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-xgar5s"
          >
            a
          </div>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-xgar5s"
          >
            g
          </div>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-xgar5s"
          >
            r
          </div>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-xgar5s"
          >
            a
          </div>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-xgar5s"
          >
            p
          </div>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-xgar5s"
          >
            h
          </div>
        </div>
        <div
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          <div
            class="MuiBox-root css-xgar5s"
          >
            s
          </div>
        </div>
      </div>
      <div
        class="MuiCardActions-root css-1h52dy8-MuiCardActions-root"
      >
        <div
          class="MuiBox-root css-8atqhb"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-hrkt27-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-1l4w6pd"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-sy308k-MuiTypography-root-MuiLink-root"
                  href="https://github.com/caponetto"
                  rel="noreferrer"
                  target="_blank"
                >
                  <button
                    aria-label="Open github"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1l55se7-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    data-testid="github-button"
                    id="github-button"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="GitHubIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-1l4w6pd"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-sy308k-MuiTypography-root-MuiLink-root"
                  href="https://twitter.com/caponetto"
                  rel="noreferrer"
                  target="_blank"
                >
                  <button
                    aria-label="Open x"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1l55se7-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    data-testid="x-button"
                    id="x-button"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="XIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M18.244 2.25h3.308l-7.227 8.26 8.502 11.24H16.17l-5.214-6.817L4.99 21.75H1.68l7.73-8.835L1.254 2.25H8.08l4.713 6.231zm-1.161 17.52h1.833L7.084 4.126H5.117z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
            >
              <div
                class="MuiBox-root css-1l4w6pd"
              >
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-sy308k-MuiTypography-root-MuiLink-root"
                  href="https://www.linkedin.com/in/ghcaponetto"
                  rel="noreferrer"
                  target="_blank"
                >
                  <button
                    aria-label="Open linkedin"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1l55se7-MuiButtonBase-root-MuiIconButton-root"
                    data-mui-internal-clone-element="true"
                    data-testid="linkedin-button"
                    id="linkedin-button"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="LinkedInIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
